src/client/providers/lintProvider.ts

```diff
@@ -114,27 +114,29 @@ export class LintProvider extends vscode.Disposable {
             });
         });
 
-        Promise.all<linter.ILintMessage[]>(promises).then(msgs => {
-            if (cancelToken.token.isCancellationRequested) {
-                return;
-            }
-
-            // Flatten the array
-            let consolidatedMessages: linter.ILintMessage[] = [];
-            msgs.forEach(lintMessages => consolidatedMessages = consolidatedMessages.concat(lintMessages));
-
-            // Limit the number of messages to the max value
-            consolidatedMessages = consolidatedMessages.filter((value, index) => index <= this.settings.linting.maxNumberOfProblems);
-
-            // Build the message and suffix the message with the name of the linter used
-            let messages = [];
-            consolidatedMessages.forEach(d => {
-                d.message = `${d.message}`;
-                messages.push(createDiagnostics(d, documentLines));
-            });
-
-            this.diagnosticCollection.delete(documentUri);
-            this.diagnosticCollection.set(documentUri, messages);
-        });
+        // linters will resolve asynchronously - keep a track of all 
+        // diagnostics reported as them come in
+        let diagnostics: vscode.Diagnostic[] = [];
+
+        promises.forEach(p => {
+            p.then(msgs => {
+                if (cancelToken.token.isCancellationRequested) {
+                    return;
+                }
+                
+                // Build the message and suffix the message with the name of the linter used
+                msgs.forEach(d => {
+                    d.message = `${d.message}`;
```

Comment:
@DonJayamanne I kept this line from the previous code but it doesn't make sense to me - as `d.message` is a `string`, surely:
```js
d.message === `${d.message}`
```
so this is a no-op?